src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1291,38 +1248,40 @@ public static string GetModulePath(string currentProcessModulePath, string hklmM
+                currentProcessModulePath += Path.PathSeparator;
+                currentProcessModulePath += GetWindowsPowerShellPSHomeModulePath();
+#endif
             }
             // EVT.Process exists
             // Now handle the case where the environment variable is already set.
             else
             {
-                // When SxS PS instance A starts SxS PS instance B, A's PSHome module path might be inherited by B. We need to remove that path from B
-                currentProcessModulePath = RemoveSxSPsHomeModulePath(currentProcessModulePath, personalModulePath, sharedModulePath, psHomeModulePath);
-
                 string personalModulePathToUse = string.IsNullOrEmpty(hkcuUserModulePath) ? personalModulePath : hkcuUserModulePath;
                 string systemModulePathToUse = string.IsNullOrEmpty(hklmMachineModulePath) ? psHomeModulePath : hklmMachineModulePath;
 
                 currentProcessModulePath = AddToPath(currentProcessModulePath, personalModulePathToUse, 0);
 
                 int insertIndex = -1;
-#if !UNIX
-                string windowsPowerShellModulePath = GetWindowsPowerShellPSHomeModulePath();
-                // If the Windows PowerShell Module path is already present, insert the system module path
-                // ($PSHOME/Modules) before it.
-                insertIndex = PathContainsSubstring(currentProcessModulePath, windowsPowerShellModulePath);
-#endif
-                currentProcessModulePath = AddToPath(currentProcessModulePath, systemModulePathToUse, insertIndex);
-            }
 
-            // if we reached this point - always add <Program Files> location to EVT.Process
-            // everything below is the same behaviour as WMF 4 code
+                insertIndex = PathContainsSubstring(currentProcessModulePath, personalModulePath);
+                if (insertIndex >= 0)
+                {
+                    insertIndex += personalModulePath.Length + 1;
+                }
 
-            // index of $PSHome\Modules in currentProcessModulePath
-            int indexOfPSHomeModulePath = PathContainsSubstring(currentProcessModulePath, psHomeModulePath);
+                currentProcessModulePath = AddToPath(currentProcessModulePath, sharedModulePath, insertIndex);
 
-            // if $PSHome\Modules not found (psHomePosition == -1) - append <Program Files> location to the end;
-            // if $PSHome\Modules IS found (psHomePosition >= 0) - insert <Program Files> location before $PSHome\Modules
-            currentProcessModulePath = AddToPath(currentProcessModulePath, sharedModulePath, indexOfPSHomeModulePath);
+                insertIndex = PathContainsSubstring(currentProcessModulePath, sharedModulePath);
+                if (insertIndex >= 0)
+                {
+                    insertIndex += sharedModulePath.Length + 1;
+                }
+
+                currentProcessModulePath = AddToPath(currentProcessModulePath, systemModulePathToUse, insertIndex);
```

Comment:
Same here, should be put in the if block.

Reply:
same as above

Reply:
Same here, `sharedModulePath` is guaranteed to be in `currentProcessModulePath` because you called `AddPath` with it.